src/combinators/mod.rs

```diff
@@ -964,6 +970,15 @@ impl<'a> Skip for io::Cursor<&'a mut [u8]> {
     }
 }
 
+impl BackToTheBuffer for &mut [u8] {
+    fn reserve_write_use<Tmp, Gen: Fn(Self) -> GenResult<(Self, Tmp)>, Before: Fn(Self, Tmp) -> GenResult<Self>>(self, reserved: usize, gen: Gen, before: Before) -> GenResult<Self> {
+        let (res, buf) = self.split_at_mut(reserved);
+        let (buf, tmp) = gen(buf)?;
+        before(res, tmp)?;
```

Comment:
You probably want to ensure here that `before()` is actually writing `reserved` bytes (by simply checking the returned slice from here), and panic otherwise. It seems problematic if less data was written (and for various impls this will cause actual problems of uninitialized data).

Reply:
Done